src/NHibernate.Test/NHSpecificTest/NH3426/Fixture.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Linq;
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.NH3426
+{
+	[TestFixture]
+	public class Fixture : TestCaseMappingByCode
+	{
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Entity>(rc =>
+			{
+				rc.Id(x => x.Id);
+				rc.Property(x => x.Name);
+			});
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		private const string id = "9FF2D288-56E6-F349-9CFC-48902132D65B";
+
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			{
+				session.Save(new Entity { Id = Guid.Parse(id), Name = "Name 1" });
+
+				session.Flush();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			using (var session = OpenSession())
+			{
+				using (var transaction = session.BeginTransaction())
+				{
+					session.Delete("from System.Object");
+
+					session.Flush();
+					transaction.Commit();
+				}
+			}
+		}
+
+		[Test]
+		public void SelectGuidToString()
+		{
+			using (var session = OpenSession())
+			{
+				var list = session.Query<Entity>()
+					.Select(x => new { Id = x.Id.ToString() })
```

Comment:
I add this test, and this test works only with SQL Server ( I tried this test on db Postgres(fixes), SQLite and MySQL too).

I think SQLite can have same problem of MySQL and Oracle (big-endian processor).

Reply:
If it does not work within a `Where`, I do not think we can consider the feature as being done. (Unless the failure is unrelated, due by example to case sensitivity or lack of `ToUpper` support. But then a better way to test the where case is needed.)

> I think SQLite can have same problem of MySQL and Oracle (big-endian processor).

Having no way to test this, since it requires a separated database server with an adequate processor architecture, we will likely not investigate further this point.